### DISCUSSION TITLE
Download URL for Gitlab releases

### DISCUSSION THREAD

**Discussion by tboerger** [renovatebot/renovate#12851] [Request Help]
Hi,

I would like to update the download URLs for Gitlab releases automatically, but I have no idea how I can get the right hash for the assetes attached to the releases. Maybe somebody got an idea how to handle that? This is what I got so far:

### .github/renovate.json

```
{
    "regexManagers": [{
        "fileMatch": [
            "defaults/main.yml"
        ],
        "matchStrings": [
            "# renovate: datasource=(?<datasource>\\S+) depName=(?<depName>\\S+)( versioning=(?<versioning>.*?))?\\n.*?_version: (?<currentValue>.*)\\s"
        ]
    }],
    "packageRules": [{
            "description": "Strip v prefix from versions",
            "groupName": "version prefix",
            "matchDatasources": [
                "gitlab-releases"
            ],
            "extractVersion": "^v(?<version>.*)$"
        }
    ]
}
```

### defaults/main.yml

```
# @var logstash_exporter_version:description: Version of the release to install
# renovate: datasource=gitlab-releases depName=alxrem/prometheus-logstash-exporter
logstash_exporter_version: 0.6.2

# @var logstash_exporter_download:description: URL to the archive of the release to install
logstash_exporter_download: "https://gitlab.com/alxrem/prometheus-logstash-exporter/uploads/2b66e95552dcd921fbbf667bb2e46cf3/prometheus-logstash-exporter-{{ logstash_exporter_version }}-linux-amd64"
```

**Comment by viceice** [comment]
I think this is currently not possible, as the hash it the id of the uploaded file, which is then linked to the gitlab release.

This can maybe resolved with a new `gitlab-release-uploads` datasource, which can implement such a feature. Not sure about this.

**Reply by tboerger** [reply]
Ok, so currently the `gitlab-releases` datasource is currently pretty useless for me :(

**Reply by rarkins** [reply]
That's unnecessarily rude. In future please be polite when interacting in this repo.

**Reply by tboerger** [reply]
That's really not meant rude. But without the ability to reference the assets of a release I can't see how the Gitlab release datasource gives value. Does it make sense to access the source code attached to a release?

I just hoped I missed something within the docs.
